1-Draft/RFCnnnn-Alias-Policy.md

```diff
@@ -0,0 +1,108 @@
+
+1. UX in interactive sessions.
+
+    Having many cmdlets and parameters with long names can make it difficult to enter them in interactive sessions.
+    Users who typically use standard settings are typically satisfied with the ```IntellySense``` capabilities.
+    On the other hand, some users actively using interactive sessions may still be annoyed by the need to type names.
+    These users should be able to upload their familiar and useful aliases sets to their interactive sessions.
+    Given the limited functionality of aliases, these users will also configure their environment by executing commands by importing their own functions and various modules using PowerShell profiles.
+    It should be noted that we are planning to add a domain-specific language (DSL). This will provide users with much greater customization capabilities than aliases.
+
+# PowerShell alias policy
+
+1. PowerShell implements standard aliases in all sessions including interactive:
+
+    * % -> ForEach-Object
+    * foreach -> ForEach-Object
+    * ? -> Where-Object
+    * where -> Where-Object
+
+    These aliases are actively used in interactive sessions and scripts.
+    ScriptAnalyzer will continue to recommend not to use them in scripts and modules.
+
+1. By default, PowerShell does not load aliases from modules and InitialSessionState.
+
+    To ensure module backward compatibility:
+    * ```Import-Module -Aliases``` - explicitly loads aliases from the module.
+    * Global configuration option ```GlobalAliasImport``` in ```powershell.config``` turns on loading aliases from modules (and InitialSessionState). It is turned off by default.
+
+1. Backward compatibility of standard aliases.
+
+    To ensure backward compatibility, three sets of aliases are created:
+    * ```Default``` - current alias set (from ```InitialSessionState```) in PowerShell Core. This is the minimum set of aliases that have no conflicts on UNIX systems.
+    * ```Extended``` - the ```Default``` set of aliases and additional popular aliases for Windows users (which can have conflicts on Unix).
+    * ```WindowsPowerShell``` - the full set of standard aliases from Windows PowerShell. This set provides full backward compatibility for Windows users.
+
+    No alias set are loaded by default.
+
+    Global configuration option ```GlobalAliasSet``` in ```powershell.config``` turns on loading the desired set. It is turned off by default.
+
+1. PowerShell prevents the use of aliases in scripts and modules.
+
+    Use of aliases is allowed only in interactive sessions.
+    Global configuration option ```GlobalAliasEnable``` in ```powershell.config``` allows the use of aliases in scripts and modules. It is turned off by default.
+
+1. Interactive session customization are delegated to users.
+
+    It is recommended to use and develop ```IntellySense``` and other means of assistance to users including ML.
```

Comment:
Please don't ask me to use ML in my interactive console. 
Users don't even wants PowerShell command usage to be exposed to Microsoft, nevermind the actual command-lines as you type.